lib/hex_web/registry_builder.ex

```diff
@@ -106,7 +107,17 @@ defmodule HexWeb.RegistryBuilder do
     :ok = :ets.tab2file(tid, String.to_char_list(file))
     :ets.delete(tid)
 
-    Application.get_env(:hex_web, :store).put_registry(File.read!(file))
+    output = File.read!(file) |> :zlib.gzip
+
+    store = Application.get_env(:hex_web, :store)
+    store.put_registry(output)
+
+    if Application.get_env(:hex_web, :signing) do
+      key = Application.get_env(:hex_web, :signing_key)
+      signature = Util.sign(output, key)
+      store.put_registry_signature(signature)
+    end
+
```

Comment:
We should compute the signature from the uncompressed registry and do it on a sha512 checksum.